wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -326,37 +343,42 @@ private MachineImpl createMachine(MachineConfigImpl machineConfig,
                                                     null);
 
         createMachineLogsDir(machineId);
-        final LineConsumer machineLogger = getMachineLogger(machineId, getMachineChannels(machine.getConfig().getName(),
-                                                                                          machine.getWorkspaceId(),
-                                                                                          machine.getEnvName())
-                .getOutput());
+        final LineConsumer machineLogger = getMachineLogger(machineId, outputConsumer);
 
         try {
+            machineRegistry.addMachine(machine);
```

Comment:
Thanks for that proposal. I do not want to change current behavior since I do believe that we have some code that rely on such behavior. I'm going to cleanup/refactoring that code later and I'll try to change that.